packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -498,6 +504,18 @@ export default function makeKernelKeeper(storage) {
     return storage.get(k);
   }
 
+  function addDynamicVatID(vatID) {
```

Comment:
I'm presuming that the plan is (eventually) for dead vats to be removed from `vat.dynamicIDs`?  Also, might it make sense to put them in a separate identifier namespace (e.g., `vd$NN`) and allocate them here, instead of passing the ID in, as that might reduce the chances of a coordination failure between allocation and use?

Reply:
Yeah, we'll remove them from that list when they're really dead. For our intermediate step, the dead vats will be remembered by a `vat.dynamicIDs` entry and a `v$NN.dead` flag (and no other `v$NN.*` keys), which will tell our startup code to create a dead-vat no-manager no-vatKeeper `ephemeral.vats` entry for it. The step after that, when we remove them completely (no object-table/etc references), will delete the two remaining DB keys, and the only trace that they ever existed will be the suspiciously-elevated `nextVatID` counter. And of course the vat will live on in the memories (messages/transcripts) of all those (surviving vats) who loved it.